Add mirror gateway document (#2105)
vdaas/vald: docs/api/mirror-gateway.md

# Vald Mirror Gateway APIs

## Overview

> Mirror Service is responsible for providing a register and an advertise interface for the Vald Mirror Gateway.

Reviewer: ```suggestion
Mirror Service is responsible for providing `Rregister` and `Advertise` interface for the Vald Mirror Gateway.
```

Reviewer 2: BTW there is a typo on ykadowak suggestion.
Rregister -> register

> Mirror Service is responsible for providing `Rregister` and `Advertise` interface for the Vald Mirror Gateway.

Author: ```suggestion
Mirror Service is responsible for providing the `Register` and `Advertise` interface for the Vald Mirror Gateway.
```
